app/tournament/Main.hs

```diff
@@ -0,0 +1,99 @@
+-- |
+-- SPDX-License-Identifier: BSD-3-Clause
+module Main where
+
+import Control.Monad.Trans.Reader (runReaderT)
+import Data.IORef (newIORef)
+import Data.Maybe (fromMaybe)
+import Network.Wai.Handler.Warp (Port)
+import Options.Applicative
+import Swarm.Game.State (Sha1 (..))
+import Swarm.Web.Tournament
+import Swarm.Web.Tournament.Database.Query
+import System.Environment (lookupEnv)
+import System.Posix.User (getEffectiveUserName)
+
+data AppOpts = AppOpts
+  { userWebPort :: Maybe Port
+  -- ^ Explicit port on which to run the web API
+  , gameGitVersion :: Sha1
+  , isLocalSocketConnection :: Bool
+  }
+
+webPort :: Parser (Maybe Int)
+webPort =
+  optional $
+    option
+      auto
+      ( long "port"
+          <> metavar "PORT"
+          <> help ("Set the web service port (or disable it with 0). Default to " <> show defaultPort <> ".")
+      )
+
+gameVersion :: Parser Sha1
```

Comment:
I don't understand why there is a flag for setting the game version.  Shouldn't it just be the hash of whatever commit the server was compiled from?  Otherwise, couldn't you just set it to something arbitrary that doesn't actually match the version that is really being used to validate and run scenarios etc.?

Reply:
Indeed, it's supposed to be the Git hash that the server was compiled from: see [this line](https://github.com/swarm-game/swarm/pull/1798/files#diff-da43e5f2be51b502f88653e55b25b54be14c32ad581bb5fa2354b2bad0eb84d8R7), which is fed into the Docker image [here](https://github.com/swarm-game/swarm/pull/1798/files#diff-d75b1260f266d4d00102549606dd159d662db8342f9d595baddeebbf2545444fR94), and then supplied as the `--version` CLI option.